### PR TITLE
Transfer tgc/cai2hcl ownership to magic-modules

### PR DESCRIPTION
**Pull request by amirkaromashkin**
This change moves tgc/cai2hcl management to magic-modules/mmv1 using new provider, which will then be used for generated cai2hcl converters.

```release-note:note
terraform-google-conversion/cai2hcl folder is now managed from GCP/magic-modules
```

### CODE REVIEW

**Comment by roaks3** [comment]
@amirkaromashkin I'm a little surprised by this change, but I may be missing context. Do you have an internal issue tracking this work, or have you been working with a member of our team on this?

**Comment by melinath** [comment]
@roaks3 I've been talking with Amir about this change - this PR should make MM take over management of cai2hcl with no diffs - other changes would ideally be in other PRs (but I believe some of these changes were necessary here as part of bringing the files under managment.)

File: mmv1/provider/terraform_tgc_cai2hcl.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2017 Google Inc.
```
Comment (roaks3) [inline mmv1/provider/terraform_tgc_cai2hcl.rb:1]: Nit: 2023

**Review by roaks3 — COMMENTED** [review]
Ok, got it, and I reviewed the internal doc. I took a first pass and things looked good, but to help with review, am I correct in assuming that `mmv1/third_party/cai2hcl` files are copied verbatim from somewhere?

**Comment by amirkaromashkin** [comment]
> Ok, got it, and I reviewed the internal doc. I took a first pass and things looked good, but to help with review, am I correct in assuming that `mmv1/third_party/cai2hcl` files are copied verbatim from somewhere?

Its not completely verbatim. Its a slightly refactored version of https://github.com/GoogleCloudPlatform/terraform-google-conversion/tree/main/cai2hcl: converters themselves were not changed, I changed packages/folder structure and helper methods like convert.go.

Testdata files copied from https://github.com/GoogleCloudPlatform/terraform-google-conversion/tree/main/testdata as is.

File: mmv1/provider/terraform_tgc_cai2hcl.rb (hunk 1 above)
Comment (amirkaromashkin) [reply mmv1/provider/terraform_tgc_cai2hcl.rb]: Done

**Comment by edisonhuang-google** [comment]
LGTM

**Comment by roaks3** [comment]
I just tried to take another look here and this is very hard to review with the assumption that everything is newly written (you mentioned that some refactoring happened). Could you please provide some more context on what was changed (and perhaps why)? Where should I be looking for changes to the previous behavior vs. parts that are copy-and-pasted?

**Comment by amirkaromashkin** [comment]
> I just tried to take another look here and this is very hard to review with the assumption that everything is newly written (you mentioned that some refactoring happened). Could you please provide some more context on what was changed (and perhaps why)? Where should I be looking for changes to the previous behavior vs. parts that are copy-and-pasted?

I guess there are 2 ways to handle it:
1. I give the broad explanation of all the changes and refactoring I made to the https://github.com/GoogleCloudPlatform/terraform-google-conversion/tree/main/cai2hcl
2. I first manually update the target repo, get it merged and then this migration will be verbatim (I'll need to give explanations there, but this will be less complicated)

I guess the 2nd option looks better. WDYT? @roaks3

**Comment by amirkaromashkin** [comment]
> > I just tried to take another look here and this is very hard to review with the assumption that everything is newly written (you mentioned that some refactoring happened). Could you please provide some more context on what was changed (and perhaps why)? Where should I be looking for changes to the previous behavior vs. parts that are copy-and-pasted?
> 
> I guess there are 2 ways to handle it:
> 
> 1. I give the broad explanation of all the changes and refactoring I made to the https://github.com/GoogleCloudPlatform/terraform-google-conversion/tree/main/cai2hcl
> 2. I first manually update the target repo, get it merged and then this migration will be verbatim (I'll need to give explanations there, but this will be less complicated)
> 
> I guess the 2nd option looks better. WDYT? @roaks3

I have just found a slight bug in a folder structure when started to implement the Option2. So I believe its  a better approach. On it.

**Comment by roaks3** [comment]
Yea, I think the second option would be ideal here. Thanks!

**Comment by roaks3** [comment]
Passing over to @melinath as I will likely be on PTO when this is ready for another look

**Comment by melinath** [comment]
https://github.com/GoogleCloudPlatform/terraform-google-conversion/pull/1416 is the PR to refactor TGC first.

**Review by melinath — CHANGES_REQUESTED** [review]
Refactor of TGC is complete - marking this as reviewed pending updates to match the final structure we settled on.

**Comment by amirkaromashkin** [comment]
> Refactor of TGC is complete - marking this as reviewed pending updates to match the final structure we settled on.

Done. Now it should produce zero diff

**Review by melinath — APPROVED** [review]
LGTM

**Pull request merged**